radio/src/bluetooth.cpp

```diff
@@ -413,6 +413,15 @@ void Bluetooth::wakeup()
     return;
   }
 
+#if defined(BT_PWR_GPIO)
+  if (g_eeGeneral.bluetoothMode == BLUETOOTH_OFF) {
+    bluetoothDisable();
+    state = BLUETOOTH_STATE_OFF;
+    wakeupTime = now + 10; /* 100ms */
+    return;
+  }
+#endif
+
```

Comment:
What was the rationale for this change? Given the immediately following code will disable it and set the wake time if it is either supposed to be OFF, or configured as trainer but trainer not enabled (although it doesn't seem to actually power .

Reply:
It looks that way, but no (hence the return statement): the BT loop assumes the module is always on (since it is hardwired this way in the radio) and runs it even if configured to off. This ensure the power port is set to off and not turned on back again for TX16s. That may be an issue for users having the internal frsky bt module, not fully sure how that module behaves

Reply:
I think I see it now, it is somehow related to the initial state, as without this check if you turn it off, it _will_ turn off , but without this check, it will always turn on at power on, rather than honour the configured "off" state. Thanks for clarifying wanted to separate between the ENable pin and PWR pin also ;)